### DISCUSSION TITLE
Bundle shape metrics

### DISCUSSION THREAD

**Discussion by wendyfyx** [dipy/dipy#2802] [1. Ideas & Feedback]
Are there bundle shape metrics (volume, length, area, diameter, etc) implemented in DIPY similar to DSI studio?

**Comment by skoudoro** [comment]
Hi @wendyfyx,

Thank you for asking. Yes, there are some metrics like `length` but I will let @BramshQamar to give you more details

**Comment by BramshQamar** [comment]
Hi @wendyfyx 

Here are some useful streamline-based functions in DIPY:

https://github.com/dipy/dipy/blob/master/dipy/tracking/utils.py
https://github.com/dipy/dipy/blob/master/dipy/tracking/streamline.py

I think we don't have functions for area and diameter but would be nice to add! 
Feel free to add new functions in your PR :)

Bramsh

**Reply by BramshQamar** [reply]
@wendyfyx 
Also, this has some useful functions for streamlines: https://github.com/dipy/dipy/blob/master/dipy/tracking/metrics.py
